driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/MongoClientImpl.java

```diff
@@ -81,8 +169,9 @@ public void close() {
     }
 
     @Override
-    public <TResult> ListDatabasesPublisher<TResult> listDatabases(final Class<TResult> clazz) {
-        return new ListDatabasesPublisherImpl<>(wrapped.listDatabases(clazz));
+    public <T> ListDatabasesPublisher<T> listDatabases(final Class<T> clazz) {
+        return PublisherCreator.createListDatabasesPublisher(null, clazz, getCodecRegistry(),
```

Comment:
Nit: statically import all of these?

Reply:
Will refactor the use of the PublisherCreator as per your suggestion